pyfr/backends/openmp/gimmik.py

```diff
@@ -28,13 +30,24 @@ def mul(self, a, b, out, alpha=1.0, beta=0.0):
             raise NotSuitableError('Matrix too dense for GiMMiK')
 
         # Generate the GiMMiK kernel
-        src = generate_mm(a.get(), dtype=a.dtype, platform='c-omp',
-                          alpha=alpha, beta=beta)
+        src = generate_mm(a.get().reshape((a.nrow, a.ncol)), dtype=a.dtype,
```

Comment:
If a reshape is needed here something has gone wrong.  We need get to provide the same shape as went in, otherwise it is bugged.

Reply:
With the new data structure all the arrays have 3 dimensions, and the first one gives the number of blocks an array has. In this case it is one, because opmats are always only one block. 
We don't need such trick with xsmm as in reality it is a contiguous 2d array. Having 1 in the beginning doesn't change anything, and c handles it as usual. With gimmik it doesn't work because gimmik tries to loop over the first dimension, and the length of the first dimension is 1.
In order to maintain compatibility I guess we need to keep arrays 3-dimensional in openmp kernel. How do you suggest solving this one? If your only concern is that this might be bugged, I tested and it works okay.

Reply:
It's a bug in the sense that when you call .get() you get something back with a different size than what you put in.  This is a change in behaviour.  If you construct a matrix on the backend with shape (r, c) and call .get() you expect to get something back of shape (r, c).  (Same with other dimensions.)  This is the case for all backends at the moment which go to great lengths to hide the internal representation.  The original shape can be recovered with ioshape.